utils/https.h

```diff
@@ -19,6 +21,22 @@ extern "C" {
  * @file utils/https.h
  */
 
+/* struct type of HTTP(S) response */
+typedef struct {
+  char* buffer; /**< Message body **/
+  size_t len;   /**< Length of message body */
+} https_response_t;
+
+/* struct type of HTTP(S) context */
+typedef struct {
+  const char* host; /**< HTTP(S) host */
+  const char* port; /**< HTTP(S) port */
```

Comment:
It is silly to use C-style string to preserve port number. Use short integer instead.